fuzzers/074-dump_all/run_fuzzer.py

```diff
@@ -0,0 +1,127 @@
+import os
+import shutil
+import sys
+import subprocess
+import signal
+from multiprocessing import Pool
+from itertools import chain
+
+# Can be used to redirect vivado tons of output
+# stdout=DEVNULL in subprocess.check_call
+#try:
+#    from subprocess import DEVNULL
+#except ImportError:
+#    import os
+#    DEVNULL = open(os.devnull, 'wb')
+
+
+# Worker function called from threads
+def start_tiles(argList):
+    blockID, start, stop = argList
+    print("Running instance :" + str(blockID))
+    subprocess.check_call(
+        "${XRAY_VIVADO} -mode batch -source $FUZDIR/jobtiles.tcl -tclargs " +
+        str(blockID) + " " + str(start) + " " + str(stop),
+        shell=True)
+
+def start_nodes(argList):
+    blockID, start, stop = argList
+    print("Running instance :" + str(blockID))
+    subprocess.check_call(
+        "${XRAY_VIVADO} -mode batch -source $FUZDIR/jobnodes.tcl -tclargs " +
+        str(blockID) + " " + str(start) + " " + str(stop),
+        shell=True)
+
+# Function called once to get the total numbers of tiles to list
+def get_nb_tiles():
+    print("Fetching total number of tiles")
+    subprocess.check_call(
+        "${XRAY_VIVADO} -mode batch -source $FUZDIR/get_tilescount.tcl",
+        shell=True)
+    countfile = open("nb_tiles.txt", "r")
+    return int(countfile.readline())
+
+def get_nb_nodes():
+    print("Fetching total number of nodes")
+    subprocess.check_call(
+        "${XRAY_VIVADO} -mode batch -source $FUZDIR/get_nodescount.tcl",
+        shell=True)
+    countfile = open("nb_nodes.txt", "r")
+    return int(countfile.readline())
+
+
+def run_pool(itemcount, nbBlocks, blocksize, nbParBlock, workFunc):
+    # We handle the case of not integer multiple of pips
+    intitemcount = blocksize * nbBlocks
+    lastRun = False
+    modBlocks = itemcount % nbBlocks
+    if modBlocks != 0:
+        lastRun = True
+        nbBlocks = nbBlocks + 1
+
+    if not os.path.exists("outdata"):
+        os.mkdir("outdata")
+
+    print(
+        "Pips Count: " + str(itemcount) + " - Number of blocks: " +
+        str(nbBlocks) + " - Parallel blocks: " + str(nbParBlock) +
+        " - Blocksize: " + str(blocksize) + " - Modulo Blocks: " +
+        str(modBlocks))
+
+    blockId = range(0, nbBlocks)
+    startI = range(0, intitemcount, blocksize)
+    stopI = range(blocksize, intitemcount + 1, blocksize)
+
+    # In case we have a last incomplete block we add it as a last
+    # element in the arguments list
+    if lastRun == True:
+        startI = chain(startI, [intitemcount])
+        stopI = chain(stopI, [itemcount])
+
+    argList = zip(blockId, startI, stopI)
+
+    with Pool(processes=nbParBlock) as pool:
+        pool.map(workFunc, argList)
+    
+    return nbBlocks
+
+def main(argv):
+    nbBlocks = 64
+    nbParBlock = 4
```

Comment:
Use argparse and make nbParBlock and nbBlocks arguments.  Then drive the arguments from the Makefile using ?=.  This will allow beefier machines to run with higher work fractions.

Reply:
Yep that was in my TODO list, but we should find a way to bench that, like detecting RAM and CPU and optimise it. What do you think ?

Reply:
For now just move the argument from hard coded in the python to a default variable in the Makefile.  This will enable easy modification using make variables, e.g:

make MAX_VIVADO_PROCESS=8 

or the like.  Using ?= allows a reasonable default.

Reply:
we could do that sure but need to tweak that differently for each main loop, they don't have the same leak. But yes could be done that way !